CMakeLists.txt

```diff
@@ -27,12 +29,36 @@ option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
 
 add_library(${PROJECT_NAME} INTERFACE)
 
+if(ENABLE_STANDALOND_WINTLS)
+  FetchContent_Declare(
+    asio
+    GIT_REPOSITORY https://github.com/chriskohlhoff/asio.git
+    GIT_TAG        asio-1-28-0
+  )
```

Comment:
Should people really opt-in to using `FetchContent`? I think it would be better if users would provide ASIO by themselves (e.g. some users prefer using git submodules).

Reply:
That might lose the out-of-the-box experience. If the user has such a need, I think they should modify the cmake file. Furthermore, WINTLS is a header-only lib, some users may not even use the cmake that comes with the project.

Reply:
I agree with @Nerixyz. I don't think it makes sense for this library to fetch Asio with CMake, as a git submodule any other way.

Instead, we should create a `FindAsio.cmake` file and use that similar to how Boost is found.

Reply:
Okay, I'll spend some time working on this. Or you can hijack this branch to modify them.

Reply:
@windowsair

I can takeover from here and do the last things needed for this to be merged to the master branch, OK?

Reply:
@laudrup 

Of course :D